### PR TITLE
Let runfiles env vars take priority

### PR DESCRIPTION
**Pull request by fmeum**
This fixes https://github.com/bazelbuild/bazel/issues/14500.

### CODE REVIEW

**Comment by phst** [comment]
I agree that this is the better choice, but I think we should try to fix https://docs.google.com/document/d/e/2PACX-1vSDIrFnFvEYhKsCMdGdD40wZRBX3m3aZ5HhVj4CtHPmiXKDCxioTUbYsDydjKtFDAzER5eg7OjJWs3V/pub first.
As an alternative, callers can pass an invalid program name using e.g. `ProgramName("/invalid")`; that's a bit of a hack, but should reliably suppress the unwanted behavior.

File: runfiles.go [hunk 1]
```diff
@@ -89,6 +89,18 @@ func New(opts ...Option) (*Runfiles, error) {
 	}
 	// See section “Runfiles discovery” in
 	// https://docs.google.com/document/d/e/2PACX-1vSDIrFnFvEYhKsCMdGdD40wZRBX3m3aZ5HhVj4CtHPmiXKDCxioTUbYsDydjKtFDAzER5eg7OjJWs3V/pub.
+	// We deviate from the doc (as of 2021-12-30) in one point to work around
```
Comment (phst) [inline runfiles.go:92]: The doc has now been updated, so I'm happy to merge this PR, but please update this comment accordingly first.

Comment (fmeum) [reply runfiles.go]: Done. I dropped everything but the link, which now does explain the process correctly.

Comment (phst) [reply runfiles.go]: Thanks!

**Pull request merged**